Source/AliveLibAO/SecurityDoor.cpp

```diff
@@ -99,5 +112,241 @@ void SecurityDoor::VScreenChanged_461F80()
     field_6_flags.Set(BaseGameObject::eDead_Bit3);
 }
 
+BOOL SecurityDoor::IsPlayerNear()
+{
+    const short xpos = FP_GetExponent(sControlledCharacter_50767C->field_A8_xpos);
+    const short ypos = FP_GetExponent(sControlledCharacter_50767C->field_AC_ypos);
+
+    if (xpos < field_10C_top_left.field_0_x || xpos > field_110_bottom_right.field_0_x)
+    {
+        return 0;
+    }
+
+    if (ypos >= field_10C_top_left.field_2_y && ypos <= field_110_bottom_right.field_2_y)
+    {
+        return 1;
+    }
+
+    return 0;
+}
+
+void SecurityDoor::VUpdate_461AD0()
+{
+    if (Event_Get_417250(kEventDeathReset_4))
+    {
+        field_6_flags.Set(Options::eDead_Bit3);
+    }
+
+    switch (field_E8_state)
+    {
+    case 0:
+        if (static_cast<int>(gnFrameCount_507670) > field_114_timer)
+        {
+            if (IsPlayerNear())
+            {
+                field_10_anim.field_4_flags.Set(AnimFlags::eBit3_Render);
+                field_E8_state = 2;
+            }
+            else
+            {
+                field_10_anim.field_4_flags.Clear(AnimFlags::eBit3_Render);
+            }
+        }
+        break;
+
+    case 1:
+        if (static_cast<int>(gnFrameCount_507670) == field_114_timer)
+        {
+            SND_SEQ_Play_477760(45u, 1, 127, 127);
```

Comment:
enum

Reply:
those don't have enums in AO yet

Reply:
Could add a place holder so the rename catches it?